### ISSUE TITLE
Agda-2.6.2.2 build failure on M1 Mac (ghc-9.2.2, cabal)

### ISSUE THREAD

**Issue by Zeta611** [agda/agda#6391]
I'm trying to install Agda, but without success:
```
❯ cabal install Agda
Resolving dependencies...
Build profile: -w ghc-9.2.2 -O1
In order, the following will be built (use -v for more details):
 - Agda-2.6.2.2 (lib:Agda, exe:agda, exe:agda-mode) (requires build)
Starting     Agda-2.6.2.2 (all, legacy fallback)
cabal: Failed to build Agda-2.6.2.2. The failure occurred during the configure
step. The build process segfaulted (i.e. SIGSEGV).
```

The log file contains little information (to me):
```
[1 of 1] Compiling Main             ( /var/folders/70/nn6j8b5x3rqcz12fq8by4hqm0000gn/T/cabal-install.-9359/dist-newstyle/tmp/src-9359/Agda-2.6.2.2/dist/setup/setup.hs, /var/folders/70/nn6j8b5x3rqcz12fq8by4hqm0000gn/T/cabal-install.-9359/dist-newstyle/tmp/src-9359/Agda-2.6.2.2/dist/setup/Main.o )
Linking /var/folders/70/nn6j8b5x3rqcz12fq8by4hqm0000gn/T/cabal-install.-9359/dist-newstyle/tmp/src-9359/Agda-2.6.2.2/dist/setup/setup ...
```

I'm using GHC 9.2.2 and cabal-install 3.6.2.0 on my M1 Mac running macOS Ventura (13.0).
Thank you!

**Comment by L-TChen** [comment]
Can you update your GHC and cabal and try again? If you are using macOS, then maybe you can just install it via homebrew?

**Comment by andreasabel** [comment]
Yes, ghc-9.2.2 might not work, more M1 fixes are in ghc-9.2.5, see: https://downloads.haskell.org/~ghc/9.2.5/docs/html/users_guide/9.2.5-notes.html

**Comment by Zeta611** [comment]
Yes, I got it working with ghc-9.2.5. Thank you!
